sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/SclService.java

```diff
@@ -328,43 +340,27 @@ public static List<ControlBlock<?>> getExtRefSourceInfo(SCL scd, ExtRefInfo extR
         if (bindingInfo.getIedName().equals(iedName)) {
             throw new ScdException("Internal binding can't have control block");
         }
-
-        String ldInst = extRefInfo.getHolderLDInst();
-        String lnClass = extRefInfo.getHolderLnClass();
-        String lnInst = extRefInfo.getHolderLnInst();
-        String prefix = extRefInfo.getHolderLnPrefix();
-        // Check holder (IED,LD,LN) exists
         SclRootAdapter sclRootAdapter = new SclRootAdapter(scd);
-        IEDAdapter iedAdapter = sclRootAdapter.getIEDAdapterByName(iedName);
-        LDeviceAdapter lDeviceAdapter = iedAdapter.getLDeviceAdapterByLdInst(ldInst)
-                .orElseThrow(
-                        () -> new ScdException(String.format(UNKNOWN_LDEVICE_S_IN_IED_S, ldInst, iedName))
-                );
-        AbstractLNAdapter<?> abstractLNAdapter = AbstractLNAdapter.builder()
-                .withLDeviceAdapter(lDeviceAdapter)
-                .withLnClass(lnClass)
-                .withLnInst(lnInst)
-                .withLnPrefix(prefix)
-                .build();
-
-        abstractLNAdapter.checkExtRefInfoCoherence(extRefInfo);
 
         // Get CBs
         IEDAdapter srcIEDAdapter = sclRootAdapter.getIEDAdapterByName(bindingInfo.getIedName());
         LDeviceAdapter srcLDeviceAdapter = srcIEDAdapter.getLDeviceAdapterByLdInst(extRefInfo.getBindingInfo().getLdInst())
                 .orElseThrow();
 
-        AbstractLNAdapter<?> srcLnAdapter = AbstractLNAdapter.builder()
-                .withLDeviceAdapter(srcLDeviceAdapter)
-                .withLnClass(extRefInfo.getBindingInfo().getLnClass())
-                .withLnInst(extRefInfo.getBindingInfo().getLnInst())
-                .withLnPrefix(extRefInfo.getBindingInfo().getPrefix())
-                .build();
-        return srcLnAdapter.getControlSetByExtRefInfo(extRefInfo);
+        List<AbstractLNAdapter<?>> aLNAdapters = new ArrayList<>();
+        aLNAdapters.add(srcLDeviceAdapter.getLN0Adapter());
+        aLNAdapters.addAll(srcLDeviceAdapter.getLNAdapters());
+
```

Comment:
This pattern is also present in LDeviceAdapter. 
It would be nice to have a method to retrieve all LN adapters including LN0. Something like `LDeviceAdapter.getLNAdaptersInclundigLN0`

Reply:
ok